code/modules/projectiles/magazines/smgs.dm

```diff
@@ -260,3 +260,28 @@
 	w_class = SIZE_SMALL
 	max_rounds = 48
 	gun_type = /obj/item/weapon/gun/smg/nailgun
+
+//-------------------------------------------------------
+//P90, a classic SMG.
+
+/obj/item/ammo_magazine/smg/p90
+	name = "\improper FN P90 magazine (5.7×28mm)"
+	desc = "A 9mm magazine for the MP5."
+	default_ammo = /datum/ammo/bullet/smg
+	caliber = "5.7×28mm"
+	icon = 'icons/obj/items/weapons/guns/ammo_by_faction/colony.dmi'
+	icon_state = "p90"
+	gun_type = /obj/item/weapon/gun/smg/p90
+	max_rounds = 50
+//-------------------------------------------------------
+//P90, a classic SMG.
+
+/obj/item/ammo_magazine/smg/p90_twe
```

Comment:
Any reason why this can't just be a variant of `/obj/item/ammo_magazine/smg/p90` e.g. `/obj/item/ammo_magazine/smg/p90/twe` so you can re-use everything a p90 has and only change whats different?

Reply:
changed now.